table.cc

```diff
@@ -2571,7 +2575,9 @@ future<row_locker::lock_holder> table::do_push_view_replica_updates(const schema
     // We'll return this lock to the caller, which will release it after
     // writing the base-table update.
     future<row_locker::lock_holder> lockf = local_base_lock(base, m.decorated_key(), slice.default_row_ranges(), timeout);
-    return lockf.then([m = std::move(m), slice = std::move(slice), views = std::move(views), base, this, timeout, time_point, source = std::move(source), &io_priority] (row_locker::lock_holder lock) mutable {
+    return utils::get_local_injector().inject("table_push_view_replica_updates_timeout", timeout).then([lockf = std::move(lockf), timeout] () mutable {
```

Comment:
This injection is also pretty ugly (and will it really compile to nothing without injections, or will it leave some do-nothing continuation), so I wish I understood what it was trying to do - in what kind of test will you use it?

Reply:
I haven't checked - it's likely to compile to nothing, since the "continuation" unconditionally returns make_ready_future<> in release builds. The intention is similar to how error injections are used all around paxos code - to check what happens when a particular view update fails due to timeout - for instance, in the dtest case I added, I checked what happens when the update times out on one node and succeeds on the other, and that reading the row later with cl=ALL is handled properly. In short - it can be useful in all scenarios which want to deterministically create a situation when a particular view update fails.